### PR TITLE
Fix 0-byte patch files

### PR DESCRIPTION
**Pull request by Hoikas**
This changeset fixes the issues with 0-byte files being downloaded from MOULa via the Gehn repair tool. It turned out that we were opening too many concurrent files. Oops! Bonus: a few code cleanups such as `HS_OVERRIDE` and `emplace_back`

### CODE REVIEW

File: Sources/Plasma/FeatureLib/pfPatcher/pfPatcher.cpp [hunk 1]
```diff
@@ -199,16 +208,14 @@ class pfPatcherStream : public plZlibStream
...
-    virtual uint32_t GetEOF() { return fOutput->GetEOF(); }
-    virtual uint32_t GetPosition() const { return fOutput->GetPosition(); }
-    virtual uint32_t GetSizeLeft() const { return fOutput->GetSizeLeft(); }
-    virtual uint32_t Read(uint32_t count, void* buf) { return fOutput->Read(count, buf); }
-    virtual void Rewind() { fOutput->Rewind(); }
-    virtual void SetPosition(uint32_t pos) { fOutput->SetPosition(pos); }
-    virtual void Skip(uint32_t deltaByteCount) { fOutput->Skip(deltaByteCount); }
+    bool AtEnd() HS_OVERRIDE { return fOutput->AtEnd(); }
+    uint32_t GetEOF() HS_OVERRIDE { return fOutput->GetEOF(); }
+    uint32_t GetPosition() const HS_OVERRIDE { return fOutput->GetPosition(); }
```
Comment (zrax) [inline Sources/Plasma/FeatureLib/pfPatcher/pfPatcher.cpp]: Was the removal of `GetSizeLeft` intentional?

File: Sources/Plasma/FeatureLib/pfPatcher/pfPatcher.cpp [hunk 2]
```diff
@@ -170,24 +170,33 @@ class pfPatcherStream : public plZlibStream
...
     {
-        fFilename = filename.Normalize();
-        return plZlibStream::Open(fFilename, mode);
+        fDLStartTime = hsTimer::GetSysSeconds();
+        if (!fOutput)
+            Open(fFilename, "wb");
     }
 
-    virtual uint32_t Write(uint32_t count, const void* buf)
+    bool Open(const plFileName& filename, const char* mode) HS_OVERRIDE
```
Comment (zrax) [inline Sources/Plasma/FeatureLib/pfPatcher/pfPatcher.cpp]: Unused parameter `filename` (we only use the stored `fFilename` below).

Comment (Hoikas) [inline Sources/Plasma/FeatureLib/pfPatcher/pfPatcher.cpp]: Required to override base implementation

File: Sources/Plasma/FeatureLib/pfPatcher/pfPatcher.cpp (hunk 1 above)
Comment (Hoikas) [inline Sources/Plasma/FeatureLib/pfPatcher/pfPatcher.cpp]: Yes, there is nothing to override, and it was unused

File: Sources/Plasma/FeatureLib/pfPatcher/pfPatcher.cpp (hunk 2 above)
Comment (Hoikas) [inline Sources/Plasma/FeatureLib/pfPatcher/pfPatcher.cpp]: I'll update it to use the parameter instead however. Could be useful in the future

**Comment by Hoikas** [comment]
Fixed unused parameter issue

**Pull request merged**